### PR TITLE
[SDEV-1556] refactor code 1

### PR DESCRIPTION
**Pull request by nandishjpatel**
Cleaned up code for the issue when sometimes calibration 2 and 3 ROC boxes pop up randomly in the canvas. Also when calibration or acquisition button are pressed disable any mouse events for calibration 2 and 3 ROC boxes.

https://github.com/delmic/odemis/pull/2504 needs to be merged first before reviewing this PR

### CODE REVIEW

File: src/odemis/gui/comp/overlay/world.py [hunk 1]
```diff
@@ -3413,7 +3414,6 @@ def _on_coordinates(self, coordinates):
         """
         if coordinates != UNDEFINED_ROI:
             self.set_physical_sel(coordinates)
-            wx.CallAfter(self.cnvs.request_drawing_update)
```
[the remark this answers is not in the document]
Comment (nandishjpatel) [reply src/odemis/gui/comp/overlay/world.py:3416]: The coordinates value is set/updated at 2 places:
- [Here](https://github.com/nandishjpatel/odemis/blob/79658df64d27db65b63140cb803f9608f656cf6f/src/odemis/gui/comp/overlay/world.py#L3563) in the function on_left_up() of FastEMROCOverlay. self.cnvs.update_drawing() method is always called at the end of on_left_up()
- [Here](https://github.com/nandishjpatel/odemis/blob/79658df64d27db65b63140cb803f9608f656cf6f/src/odemis/gui/cont/project.py#L501) in function _on_button() of FastEMCalibrationRegionsController. fit_view_to_bbox() method called after this requests a drawing update.
- [Here](https://github.com/nandishjpatel/odemis/blob/79658df64d27db65b63140cb803f9608f656cf6f/src/odemis/gui/cont/project.py#L506) in function _on_button() of FastEMCalibrationRegionsController. remove_calibration_overlay() called after this requests a drawing update.

Hence I removed the call.

File: src/odemis/gui/comp/overlay/world.py [hunk 2]
```diff
@@ -3512,22 +3512,25 @@ def draw(self, ctx, shift=(0, 0), scale=1.0):
...
         cnvs (FastEMAcquisitionCanvas): canvas for the overlay
         coordinates (TupleContinuousVA): VA representing region of calibration coordinates
         label (str or int): label to be displayed next to rectangle
+        sample_bbox (tuple): bounding box coordinates of the sample holder (minx, miny, maxx, maxy) [m]
         colour (str): hex colour code for ROC display in viewport
         """
         super(FastEMROCOverlay, self).__init__(cnvs, coordinates, colour)
         self.label = label
+        self._sample_bbox = sample_bbox
+        self.enable = True
```
[the remark this answers is not in the document]
Comment (nandishjpatel) [reply src/odemis/gui/comp/overlay/world.py:3526]: > Maybe you could have instead a active VA and selected VA?

Makes sense, thanks

File: src/odemis/gui/comp/overlay/world.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pieleric) [reply src/odemis/gui/comp/overlay/world.py:3417]: Ok, so it's not solving any specific issue? If so, I'd suggest to leave it there, as in the future, if there is a new caller, it doesn't have to be careful to also update the drawing. 
Please note that `request_drawing_update()` is very light. It just sets a flag to indicate that soon the canvas should be redrawn. So it doesn't save much to remove it.

[the remark this answers is not in the document]
Comment (nandishjpatel) [reply src/odemis/gui/comp/overlay/world.py:3417]: > Ok, so it's not solving any specific issue?
There was a bug where calibration 2 and 3 boxes pop-up unexpectedly. Removing the request drawing update in the callback makes it on the safer side of things. Anyone who set/update the coordinates has to request a drawing update. Does it make sense?

[the remark this answers is not in the document]
Comment (pieleric) [reply src/odemis/gui/comp/overlay/world.py:3417]: Well, redrawing  the canvas can happen for many reasons. Resizing the window, zooming in/out the view, dragging the view, a new image... all this trigger a redrawing. If the model is wrong and this leads to RoC appearing, not drawing it immediately is very unlikely to fix it properly.
So I'd rather keep the standard behaviour of the "controller" (as in MVC): if the model updates, the controller takes care of updating the view.

[the remark this answers is not in the document]
Comment (nandishjpatel) [reply src/odemis/gui/comp/overlay/world.py:3417]: Understood, then I revert back the change

**Review by tepals — APPROVED** [review]

**Review by pieleric — APPROVED** [review]

**Pull request merged**